docs/diagnostics/PX1063.md

```diff
@@ -0,0 +1,117 @@
+# PX1063
+This document describes the PX1063 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                         | Type                           | Code Fix    | 
+| ------ | ----------------------------------------------------------------------------------------- | ------------------------------ | ----------- | 
+| PX1063 | The declaration of the exception class does not contain a serialization constructor. This will cause an incorrect deserialization of the exception data and will lead to runtime errors in Acumatica ERP. | Error | Available | 
+
+## Diagnostic Description
+
+Every class that that is derived from the `System.Exception` class (including PXException-derived classes) must implement the serialization constructor.
+If the serialization constructor is not defined, it will lead to the loss of exception data during serialization or deserialization and also can lead to runtime errors, especially when Acumatica ERP is used in the cluster mode.
+
+The code fix generates the serialization constructor for the exception class. The generated constructor will be private for sealed exception classes and protected for all other classes.  The generated constructor will call the base serialization constructor. 
+If the exception class declares a new serializable data then the constructor will contain a call to the `RestoreObjectProps` static method of the `PX.Common.ReflectionSerializer` class. This class is responsible for Acumatica custom reflection-based serialization and deserialization of data. If the exception class does not declare new serializable data, the serialization constructor body will be empty.
+
+By serializable data, the diagnostic understands any new data declared in the exception class that will be serialized by .Net serializers and Acumatica reflection based serializer which are the following:
+ - New instance fields if they are not marked with `System.NonSerializedAttribute` attribute
+ - New auto generated instance properties if they are not marked with `System.NonSerializedAttribute` attribute
+For example, the following cases are not considered serializable:
+ - Static fields and constants
+ - Static properties
+ - Calculated properties
+ - Properties with explicit backing field as shown in the following example
+	```C#
+	public int Property 
+	{
+	   get =>  _fieldFromBaseClass;
+	   set 
+	   {
+		   _fieldFromBaseClass = value;
+	   }
+	}
+	```
+ - Fields and properties marked as non serializable as shown in the following example:
+	```C#
+	[field: NonSerialized]
+	public int Property { get; set; } 
+	```
+
+## Example of Incorrect Code for a Class Without Serializable Data
+
+```C#
+public class DocumentNotApprovedException : PXException
+{
+	public int Property => 4;
```

Comment:
These code example became incorrect since the code won't compile. There are three properties with the same name `Property`.  Imho, it's unnecessary to cover all non serializable cases here, just keep the 
```csharp
[field: NonSerialized]
public int Property { get; set; } 
```
and delete the rest from the example. It would be good to provide a property more close to the real use cases like:
```csharp
[field: NonSerialized]
public Guid? DocumentNoteID { get; set; } 
```